transport/transporttest/context.go

```diff
@@ -54,22 +78,32 @@ func NewContextMatcher(t *testing.T, ttl time.Duration) *ContextMatcher {
 func (c *ContextMatcher) Matches(got interface{}) bool {
 	ctx, ok := got.(context.Context)
 	if !ok {
-		c.t.Logf("no context")
+		c.t.Logf("expected a Context but got a %T: %v", got, got)
 		return false
 	}
 
-	d, ok := ctx.Deadline()
-	if !ok {
-		c.t.Logf("no context deadline")
-		return false
+	if c.ttl != 0 {
+		d, ok := ctx.Deadline()
+		if !ok {
+			c.t.Logf(
+				"expected Context to have a TTL of %v but it has no deadline", c.ttl)
+			return false
+		}
+
+		ttl := d.Sub(time.Now())
+		maxTTL := c.ttl + c.TTLDelta
+		minTTL := c.ttl - c.TTLDelta
+		if ttl > maxTTL || ttl < minTTL {
+			c.t.Logf("TTL out of expected bounds: %v < %v < %v", minTTL, ttl, maxTTL)
+			return false
+		}
 	}
 
-	ttl := d.Sub(time.Now())
-	maxTTL := c.ttl + c.TTLDelta
-	minTTL := c.ttl - c.TTLDelta
-	if ttl > maxTTL || ttl < minTTL {
-		c.t.Logf("TTL out of expected bounds: %v < %v < %v", minTTL, ttl, maxTTL)
-		return false
+	if c.baggage != nil {
+		headers := baggage.FromContext(ctx)
+		if !assert.Equal(c.t, c.baggage, headers, "context baggage did not match") {
```

Comment:
we should be consistent in how we handle mismatches here. We do `assert.Equal` here, but above, we do a `t.Logf`. `assert.Equal` will fail the test, while `Logf` does not.